### ISSUE TITLE
Excluded modules still included by apidoc

### ISSUE THREAD

**Issue by atodorov** [sphinx-doc/sphinx#4581]
Subject: I have a command which excludes some modules and that works with Sphinx 1.6.6 but fails with 1.6.7

### Problem
I think the issue I'm seeing was caused by https://github.com/sphinx-doc/sphinx/pull/4449

#### Procedure to reproduce the problem
```
git clone https://github.com/kiwitcms/Kiwi
cd Kiwi/
pip install -r requirements/devel.txt 
pip install -r requirements/mysql.txt
pip install --update Sphinx ## to make sure we're using 1.6.7
cd docs/
make apidoc
git diff
```

The Makefile specifies:

```
apidoc:
        $(SPHINXAPIDOC) -o source/modules/ -f -e -M -T ../ ../manage.py ../setup.py \
                        ../*/tests.py ../*/tests/*.py ../*/test_*.py \
                        ../settings/test/*.py ../*/migrations/*.py ../tcms_api/setup.py
```

#### Expected results

Excluded modules are not included by apidoc

### Reproducible project / your project
- https://github.com/kiwitcms/Kiwi

### Environment info
- OS: Linux, RHEL 7.4
- Python version: 3.6
- Sphinx version: 1.6.7

**Comment by tk0miya** [comment]
Note: I reproduced this with following Dockerfile:
```
FROM tk0miya/sphinx-html

RUN git clone https://github.com/kiwitcms/Kiwi
WORKDIR /docs/Kiwi/
RUN apt install -y mysql-client libmysqlclient-dev
RUN pip3 install -r requirements/devel.txt_
RUN pip3 install -r requirements/mysql.txt
RUN pip3 install -U Sphinx ## to make sure we're using 1.6.7
WORKDIR /docs/Kiwi/docs/
RUN make apidoc
RUN git diff
```

**Comment by tk0miya** [comment]
Fixed by #4592 
Thank you for reporting!
